borg/archiver.py

```diff
@@ -40,18 +41,56 @@
 DASHES = '-' * 78
 
 
-class ToggleAction(argparse.Action):
-    """argparse action to handle "toggle" flags easily
+def argument(args, str_or_bool):
+    """If bool is passed, return it. If str is passed, retrieve named attribute from args."""
+    if isinstance(str_or_bool, str):
+        return getattr(args, str_or_bool)
+    return str_or_bool
 
-    toggle flags are in the form of ``--foo``, ``--no-foo``.
 
-    the ``--no-foo`` argument still needs to be passed to the
-    ``add_argument()`` call, but it simplifies the ``--no``
-    detection.
+def with_repository(fake=False, create=False, lock=True, exclusive=False, manifest=True, cache=False):
```

Comment:
The defaults are what is used most commonly:

- 8 commands use the defaults (cache=False, manifest=True, exclusive=False)
- 5 have manifest=False (init, check and all do_debug stuff)
- 4 use cache=True
- 4 use exclusive=True